Morphic/Morphic.tex

```diff
@@ -593,69 +593,57 @@ \subsection{Morphic animations}
-				initialAnswer: 'no name'].
-	]
-	\end{ExecuteSmalltalkScript}
+SBEScreenshotRecorder writeTo: 'figures/dialog.png' building: [:helper |
+	helper runModal: [
+		Project uiManager
+			request: 'What''s your name?'
+			initialAnswer: 'no name'].
+]
+\end{ExecuteSmalltalkScript}
 \begin{figure}[htb]
-\begin{minipage}{0.55\textwidth}
-	\ifluluelse
-		{\centerline{\includegraphics[scale=0.65]{dialog}}}
-		{\centerline{\includegraphics[width=5cm]{dialog}}}
-	\caption{Dialog displayed by \ct{FillInTheBlank request: 'What''s your name?' initialAnswer: 'no name'}.
+\begin{minipage}{0.363\textwidth}
+	\centerline{\includegraphics[width=\textwidth]{dialog}}
+	\caption{Dialog displayed by \ct{UIManager>>>request:initialAnswer:}.
 		\label{fig:dialogName}}
 \end{minipage}
 \hfill
 \begin{ExecuteSmalltalkScript}
 SBEScreenshotRecorder writeTo: 'figures/popup.png' building: [:helper |
-	| menu |
-	menu := PopUpMenu
-		labelArray: #('circle' 'oval' 'square' 'rectangle' 'triangle')
-		lines: #(2 4).
-	helper
-		runModal: [menu startUpWithCaption: 'Choose a shape']
-		then: [helper closeBalloons].
+	helper runModal: [
+		Project uiManager
+			chooseFrom: #('circle' 'oval' 'square' 'rectangle' 'triangle')]
 ]
 \end{ExecuteSmalltalkScript}
-\begin{minipage}{0.38\textwidth}
+\begin{minipage}{0.587\textwidth}
 	\vfill
-	\ifluluelse
-		{\centerline{\includegraphics [scale=0.65]{popup}}}
-		{\centerline{\includegraphics[width=3cm]{popup}}}
+	\centerline{\includegraphics[width=\textwidth]{popup}}
 	\vfill
 	\vspace{4ex}
-	\caption{Pop-up menu displayed by \ct{PopUpMenu>>>startUpWithCaption:}.}
+	\caption{Dialog displayed by \ct{Project uiManager chooseFrom: \#('circle' 'oval' 'square' 'rectangle' 'triangle')}.}
 \end{minipage}
 \end{figure}
 
-%\begin{figure}[ht]
-%	\centerline{\includegraphics[width=5cm]{dialog}}
-%	\caption{Dialog displayed by \ct{FillInTheBlank request: 'What''s your name?' initialAnswer: 'no name'}.
-%		\label{fig:dialogName}}
-%\end{figure}
+To prompt the user for input, the \clsind{UIManager} class provides a few ready-to-use dialog boxes.
+For instance, the \mthind{UIManager}{request:initialAnswer:} method returns the string entered by the user (\figref{dialogName}).
+However, \clsind{UIManager} is an abstract class that has several subclasses such as \clsind{MorphicUIManager}, \clsind{MVCUIManager}, and others.
+What's the reason for this?
+
+\sq supports a wide range of different UI systems of which Morphic is only one of many.
+For instance, another UI system in \sq is \emph{MVC} (named after its tripartite Model View Controller architecture) which was the only available UI system in classic Smalltalk-80.
+To allow developers to write dialogs that work for any UI system, \sq provides the \clsind{UIManager} facade that defines a consistent interface to invoke user dialogs for any UI system.
+Every UI system then is represented by a subinstance of the class \clsind{Project} which instantiates a customized subclass of the \clsind{UIManager}.
```

Comment:
remove facade, to specific for an introductory book

Reply:
719dc16